src/hooks/core/useCoreChatSystemHook.ts

```diff
@@ -178,21 +322,33 @@ const useCoreChatSystemHook = () => {
   };
 
   React.useEffect(() => {
-    if (!lastJsonMessage && !localDb) return;
+    if (!lastAnonymousMessage || !localDb) return;
 
-    const {type} = lastJsonMessage;
+    const {type} = lastAnonymousMessage;
     if (type === 'health.check') return;
     if (type === 'notification.message_new') {
-      saveChannelListData().catch((e) => console.log(e));
+      saveChannelListData(lastAnonymousMessage, ANONYMOUS).catch((e) => console.log(e));
     }
-  }, [lastJsonMessage, localDb]);
+  }, [JSON.stringify(lastAnonymousMessage), localDb]);
 
   React.useEffect(() => {
-    if (isEnteringApp) {
-      getAllAnonymousPostNotifications().catch((e) => console.log(e));
+    if (!lastSignedMessage || !localDb) return;
+    const {type} = lastSignedMessage;
+    if (type === 'health.check') return;
+    if (type === 'notification.message_new' || type === 'notification.added_to_channel') {
+      saveChannelListData(lastSignedMessage, SIGNED).catch((e) => console.log(e));
+    }
+  }, [JSON.stringify(lastSignedMessage), localDb]);
+
+  React.useEffect(() => {
+    const isResetNav = params?.isReset;
+    if (isEnteringApp && migrationStatus === 'MIGRATED' && !isResetNav) {
+      getAllSignedChannels().catch((e) => console.log(e));
+      getAllSignedPostNotifications().catch((e) => console.log(e));
       getAllAnonymousChannels().catch((e) => console.log(e));
+      getAllAnonymousPostNotifications().catch((e) => console.log(e));
     }
-  }, [localDb, isEnteringApp]);
+  }, [isEnteringApp, migrationStatus]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the two `React.useEffect` hooks, you're stringifying `lastAnonymousMessage` and `lastSignedMessage` on every render, even though they might not have changed. This could lead to unnecessary re-renders and performance issues. Consider using a memoized version of these values with `React.useMemo` or storing the stringified versions in the component's state with `React.useState` and only updating them when the original values change.


<!-- This is an auto-generated comment by OSS CodeRabbit -->